hack/local-up-karmada-helm.sh

```diff
@@ -0,0 +1,183 @@
+PULL_MODE_MEMBERS["member2"]="${HOME}/.kube/members.config"
+
+echo "########## start installing karmada control plane ##########"
+
+# 1. create KinD cluster if you set NEED_CREATE_KIND_CLUSTER
+if ${NEED_CREATE_KIND_CLUSTER}; then
+  # 1.1 create karmada-host cluster
+  kind delete clusters karmada-host
+  rm -rf ~/.karmada "${KARMADA_HOST_KUBECONFIG}"
+  hack/create-cluster.sh ${KARMADA_HOST_NAME} "${KARMADA_HOST_KUBECONFIG}"
+fi
+
+# 2. fetch images in advance is you set IMAGE_FROM
+if [ "${IMAGE_FROM}" == "pull-in-advance" ]; then
+  ## 2.1 use 'docker pull' to fetch target images to local node in advance
+  imgs=$(cat charts/karmada/values.yaml | grep -C 1 'repository:' | sed 's/*karmadaImageVersion/latest/g' | awk -F ':' '{print $2}' | sed 's/\"//g' | xargs -n3 | awk '{print $1"/"$2":"$3}')
+  for img in ${imgs}; do
+    docker pull "${img}"
+  done
+elif [ "${IMAGE_FROM}" == "make" ]; then
+  ## 2.2 build karmada images from source code with latest tag and pull images (other than karmada) in advance
+  export VERSION="latest"
+  export REGISTRY="docker.io/karmada"
+  make images GOOS="linux" .
+
+  imgs=$(cat charts/karmada/values.yaml | grep -v 'karmada' | grep -C 1 'repository: ' | sed 's/*karmadaImageVersion/latest/g' | awk -F ':' '{print $2}' | sed 's/\"//g' | xargs -n3 | awk '{print $1"/"$2":"$3}')
+  for img in ${imgs}; do
+    docker pull "${img}"
+  done
+fi
+
+# 3. load images into KinD karmada-host cluster in advance if you set LOAD_IMAGE_IN_ADVANCE
+if ${LOAD_IMAGE_IN_ADVANCE}; then
+  imgs=$(cat charts/karmada/values.yaml | grep -C 1 'repository:' | sed 's/*karmadaImageVersion/latest/g' | awk -F ':' '{print $2}' | sed 's/\"//g' | xargs -n3 | awk '{print $1"/"$2":"$3}')
+  for img in ${imgs}; do
+    kind load docker-image "${img}" --name ${KARMADA_HOST_NAME}
+  done
+fi
+
+# 4. this script try to deploy karmada-apiserver by host-network
+# so, it needs to get host-network ip (node ip) from kube-apiserver, and then add this ip to values.yaml as SANs of certificate
+export KUBECONFIG=${KARMADA_HOST_KUBECONFIG}
+HOST_IP=$(kubectl get ep kubernetes -o jsonpath='{.subsets[0].addresses[0].ip}')
+sed -i'' -e "/localhost/p; s/localhost/${HOST_IP}/" charts/karmada/values.yaml
+
+# 5. install karmada at karmada-host cluster by helm
+helm install karmada -n karmada-system \
+        --kubeconfig "${KARMADA_HOST_KUBECONFIG}" \
+        --create-namespace \
+        --dependency-update \
+        --set apiServer.hostNetwork=true,components={"search,descheduler"} \
+        ./charts/karmada
+
+# 6. export kubeconfig of karmada-apiserver to local path
+kubectl get secret -n karmada-system karmada-kubeconfig -o jsonpath={.data.kubeconfig} | base64 -d > ~/.kube/karmada-apiserver.config
+KARMADA_APISERVER_ADDR=$(kubectl get ep karmada-apiserver -n karmada-system | tail -n 1 | awk '{print $2}')
+sed -i'' -e "s/karmada-apiserver.karmada-system.svc.cluster.local:5443/${KARMADA_APISERVER_ADDR}/g" ~/.kube/karmada-apiserver.config
```

Comment:
What happen when cluster dns domain is not `cluster.local`

Reply:
you got the point, I also think it is not fit to set `*.cluster.local` to kubeconfig of karmada-apiserver, but now our installation just set `karmada-apiserver.karmada-system.svc.cluster.local:5443` as apiserver address in kubeconfig of karmada-apiserver by default. I described my confusing as #4013

I still can not figure why we use `karmada-apiserver.karmada-system.svc.cluster.local:5443`, I never succeed in using default karmada-apiserver kubeconfig, so, I changed it to ip.

Reply:
Hi @liangyuanpeng, any more questions?

Reply:
@chaosi-zju
When the cluster domain is `my.k8s` and this is not working, right? So it's need to match `karmada-apiserver.karmada-system.svc.my.k8s` and not `karmada-apiserver.karmada-system.svc.cluster.local` 

Remind me if i missed something,thanks.

Reply:
We can finish it in other PR.

Reply:
> @chaosi-zju When the cluster domain is `my.k8s` and this is not working, right? So it's need to match `karmada-apiserver.karmada-system.svc.my.k8s` and not `karmada-apiserver.karmada-system.svc.cluster.local`

I think what you said makes perfect sense~

So, I updated to `sed -i'' -e "s/karmada-apiserver.karmada-system.svc.*:5443/${KARMADA_APISERVER_ADDR}/g" "${KARMADA_APISERVER_KUBECONFIG}"`.

just change `cluster.local` to `*`.